docs/design.md

```diff
@@ -0,0 +1,131 @@
+Privacy and security are critical.
+* **No need for high performance**. Unless it results in an awful user experience
+* **Generalization**. We want the virtual coach system as well as individual components to be reusable outside of the 
+PerfectFit project. So individual components should be modular, and Niceday should be swappable with any other chat 
+framework.
+
+## Functional requirements
+See [this overview](https://nlesc.sharepoint.com/:x:/s/team-flow/EUCPwrKzPY1BikrYk_BDhCcB8vBLgsI-xsT63dedExPQaA?e=GlfxeO)
+
+---
+
+## Architecture sketch
+<img src = "img/design.png" width = "1000" title="Design">
+See it also on [whimsical](https://whimsical.com/perfectfit-UtvRnxdP8P79humXTnjb9J)
+
+---
+## Components
+### Smartphone applications
+#### Niceday app (already developed)
+The user interacts with the
+[NiceDay smartphone app](https://play.google.com/store/apps/details?id=nl.sense.goalie2&hl=nl&gl=US).
+This app is developed by [sense-health](https://sense-health.com/) which is part of the consortium.
+
+It's main features are:
+* Trackers: simple panels that track amongst others: mood, number of cigarettes smoked, step count, thought records.
+* Daily planner: overview of planned actions
+* Messaging
+ 
+#### Sensor data collection app (out of scope)
+A separate app for collecting sensor data that is needed to estimate physical capacity. 
+* The Niceday app itself cannot gather this data.
+* At the start of an activity the sensor data collector is triggered to start recording sensor data.
+* After an activity or upon Wifi connection the data is pushed to the PerfectFit sensor data REST API.
+* Will be based on an existing app and developed outside of scope of PerfectFit
+virtual coach system.
+--- 
+
+### Niceday components
+#### Niceday server (already developed)
+* Backend for the Niceday app.
+* Data storage: 
+  - user profile data, 
+  - app-interaction data
+  - messages
+* Handles authentication
+
+--- 
+### PerfectFit virtual coach system components
+We cannot change much in the NiceDay app, new PerfectFit features will be implemented 
+through the virtual coach that interacts with the user through the messaging function of the 
+NiceDay app. We will make use of Niceday trackers to gather data, 
+and make use of the Niceday daily planner to plan activities.
+
+
+### Sensor data REST API
+A Sensor REST API that allows for the sensor data collection app to send data to PerfectFit system.
+* Writes data directly to the database using the ORM
+
+### Rasa agent
+The conversational agent developed in the [Rasa](https://rasa.com/) framework.
+* Given an input message of the user, perform the appropriate action.
+* Actions can range from a simple text message response to something more complex, such as:
+  - using an algorithm (through importing the library) 
+  - querying the perfectfit database using ORM
+  - calling the niceday-api for information from niceday-server
```

Comment:
Apologies for my ignorance, as I'm sure this was discussed in another meeting, but I'm now a bit confused by the Rasa agent communicating with the Niceday server using both the Rasa agent REST API (via niceday-broker) and seemingly directly via the 'niceday-api'.

I see from the descriptions below that both the niceday-broker and the niceday-api are used for different interactions, but why are they not a single component? Both are REST APIs written in JS and using goalie-js. Or are they in fact the same?  The architecture diagram seems to indicate these are two totally separate interaction methods.

Reply:
No, good point! 

They are totally separate. And the broker is actually more a worker, it is not a REST api, in fact it has no interface. It just forwards messages back and forth. So it has that specific role. I've added that it is a worker in https://github.com/PerfectFit-project/virtual-coach-server/pull/49/commits/c11beabfae0e6af42c2407c351993ca59b028d35 . But I would like to ask you if you can think of how to make it clearer? If you want you can directly commit a change.

The niceday-api is for all other functionality we might need from niceday.